### ISSUE TITLE
[BUG] conditional view step in controller is not processed … er … conditionally

### ISSUE THREAD

**Issue by djbpitt** [eXist-db/exist#4513]
**Describe the bug**

Using a pipeline managed by *controller.xql* I want most queries to pass through two view steps, the first to convert the model to an HTML fragment and the second to wrap the fragment in a template that contributes the HTML boilerplate (root element, head). (Please don’t ask why I’m not using eXist-db built-in templating; my question is about understanding how sequential view steps work in the controller, and not about other ways to get the result that I’m after.) But I also execute a few ajax calls, for which I want to return the HTML fragment without the wrapper, and those can be distinguished from the regular situation because I’ve given ajax modules and the associated first view steps filenames that begin with the string ‘ajax-’. I think I should be able to make the execution of the second view step conditional on the filename (expressed as `$exist:path`), but that doesn’t seem to work for me.

A minimal working (that is, non-working) example is available at https://github.com/djbpitt/controller-view-pipeline. The view part of the controller reads:

```
<view>
    (:transformation to html is different for different modules:)
    <forward servlet="XSLTServlet">
        <set-attribute name="xslt.stylesheet" value="{concat($exist:root, $exist:controller, '/views/', $exist:path, '-to-html.xsl')}"/>
        <set-attribute name="xslt.fqcontroller" value="{$fqcontroller}"/>
    </forward>
{if (starts-with($exist:path, 'ajax-')) then () else
    <forward servlet="XSLTServlet">
        <set-attribute name="xslt.stylesheet" value="{concat($exist:root, $exist:controller, '/views/wrapper.xsl')}"/>
        <set-attribute name="xslt.fqcontroller" value="{$fqcontroller}"/>
    </forward>}
</view>
```

**Expected behavior**

I expect http://localhost:8080/exist/apps/controller-view-pipeline/read?id=GH-TIMES-18040106 to be passed through *wrapper.xsl*, and it is. I expect http://localhost:8080/exist/apps/controller-view-pipeline/ajax-read?id=GH-TIMES-18040106 **not** to be passed through *wrapper.xsl*, but it, too, gets wrapped. That is, I expect the XQuery *if-then-else* that is wrapped around the second view step to cause that step to be ignored for URLs where `$exist:path` begins with the string 'ajax-'.

**To Reproduce**

Clone https://github.com/djbpitt/controller-view-pipeline, build, install, and run http://localhost:8080/exist/apps/controller-view-pipeline/ajax-read?id=GH-TIMES-18040106.

**Context (please always complete the following information):**
 - OS: macOS 12.5
 - eXist-db version: 6.0.1
 - Java Version: 1.8.0_311

**Additional context**
- How is eXist-db installed? DMG
- Any custom changes in e.g. `conf.xml`? No

**Comment by djbpitt** [comment]
User error: `$exist:path` begins with a slash, and what begins with 'ajax-' is `$exist:resource`. Apologies for the noise.

**Comment by line-o** [comment]
Still interesting use case that you showed here. I did not know that this is possible to chain forward requests.

**Comment by djbpitt** [comment]
@line-o I learned about chaining XSLT forward requests from the eXist-db book (example on p. 206); I see now that it's also at http://exist-db.org/exist/apps/doc/urlrewrite.xml?field=all&id=D3.29#D3.29. If chaining XQuery forward requests is documented anywhere, though, I couldn't find it, and I stumbled a lot before I figured out that `request:get-data()` would collect the forwarded output for subsequent processing by a downstream XQuery (not XSLT) view.
